src/vivarium_cluster_tools/psimulate/runner.py

```diff
@@ -348,10 +348,12 @@ def write_results_batch(ctx: RunContext,
             results_to_write.to_hdf(temp_output_path, 'data')
             temp_output_path.replace(output_path)
             break
-        except Exception:
+        except Exception as e:
             logger.warning(f'Error trying to write results to hdf, retries remaining {retries}')
             sleep(30)
             retries -= 1
+            if not retries:
+                logger.warning(f"Retries exhausted: {e}")
```

Comment:
I would log a warning here that says the filesystem is unreachable, and then actually raise the error.  Otherwise, you're continuing your psimulate run even though you have failed to write a batch of results, or something bad has happened when you're replacing the existing file, or some some other i/o related shenanigans. You're also very likely in a place (given that you've already done retries) where the next batch of results is going to have the same errors.